yarn-project/merkle-tree/src/indexed_tree/indexed_tree.ts

```diff
@@ -227,10 +258,18 @@ export class IndexedTree implements MerkleTree {
 
   /**
    * Saves the initial leaf to this object and saves it to a database.
+   * TODO: what will the size be
    */
-  private async init() {
+  private async init(initialSize = 0) {
+    // TODO: increase the initial size of the tree to the size of a full rollup insertion - change reflected in c++ to allow subtree insertion
+
     this.leaves.push(initialLeaf);
     await this.underlying.appendLeaves([hashEncodedTreeValue(initialLeaf, this.hasher)]);
+
+    // TODO: optimise
+    for (let i = 1; i < initialSize; i++) {
+      await this.appendLeaf(Buffer.from([i]));
```

Comment:
Shouldn't we be inserting empty leaves, as opposed to incremental values?

Reply:
This will be empty leaves once the merkle tree impl has been updated to have a zero case that isnt hash(0,0,0). Ill create a ticket for this.